presto-hive-common/src/main/java/com/facebook/presto/hive/BlockLocation.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.hive;
+
+import com.facebook.drift.annotations.ThriftConstructor;
+import com.facebook.drift.annotations.ThriftField;
+import com.google.common.collect.ImmutableList;
+
+import javax.annotation.Nullable;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static java.util.Objects.requireNonNull;
+
+public class BlockLocation
+{
+    private final List<String> hosts;
+    private final long offset;
+    private final long length;
+
+    @ThriftConstructor
+    public BlockLocation(List<String> hosts, long offset, long length)
+    {
+        this.hosts = requireNonNull(hosts, "hosts is null");
+        this.offset = offset;
+        this.length = length;
+    }
+
+    public BlockLocation(org.apache.hadoop.fs.BlockLocation blockLocation)
+    {
+        requireNonNull(blockLocation, "blockLocation is null");
+        try {
+            this.hosts = ImmutableList.copyOf(blockLocation.getHosts());
+        }
+        catch (IOException e) {
```

Comment:
a regular IOException is preferable

Reply:
what do you mean by regular IOException ? this is java.io.IOException .. the same is used by apache hadoop.

Reply:
Yes, use the IOException instead of UncheckedIOException. No need to throw anything new here.

Reply:
this is a good point. i realize you just moved this code from InternalHiveSplitFactory, but why are we throwing an UnchecekdIOException? I feel like we should either actually handle the error and throw an appropriate PrestoException, or we should propagate the original exception so the caller knows they should handle it.